.github/workflows/maven-node-build.yaml

```
@@ -0,0 +1,38 @@
+name: build all
+
+on: [pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      TZ: Europe/Berlin # timezone
+    strategy:
+      fail-fast: false
+      matrix:
+        include: # hier müssen die Pfade angegeben werden
+          - app-path: refarch-backend
+          - app-path: refarch-frontend
+          - app-path: refarch-eai
+          - app-path: refarch-webcomponent
+    steps:
+      - uses: actions/checkout@v4
+      - if:  ${{hashFiles(format('./{0}/package.json', matrix.app-path))!=null}}
+        id: node
+        uses: it-at-m/.github/.github/actions/action-npm-build@main
+        with:
+          app-path:  "${{ matrix.app-path }}"
+      - if:  ${{hashFiles(format('./{0}/pom.xml', matrix.app-path))!=null}}
+        id: maven
+        uses: it-at-m/.github/.github/actions/action-maven-build@main
+        with:
+          java-version: 21
+          app-path: "${{ matrix.app-path }}"
+      - if: github.ref == 'refs/heads/main'
+        uses: it-at-m/.github/.github/actions/action-build-image@main
+        with:
+          path: "${{  matrix.app-path }}"
+          image-name: "${{  matrix.app-path }}"
+          artifact-name: ${{join(steps.*.outputs.artifact-name)}}
+          registry-password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ghcr.io
+          registry-username: ${{ github.actor }}
```

```suggestion
          registry-username: ${{ github.actor }}
          image-tags: |
            type=raw,value=dev
```

[Default image tag is latest](https://github.com/it-at-m/.github/blob/main/.github/actions/action-build-image/action.yml#L15-L19) what doesn't really make sense for merging on main.
We should probably discuss what the correct lable for main is. I still like `dev` the most.

---

I discussed this with @DanielOber again and as you can see currently this workflow only runs on PRs and not on the main branch. We could build a `dev` image on PRs (however this means different PRs would compete for the `dev` tag). And then only release the `latest` image when we have a push on `main`.
@DanielOber since we agreed we only want to run linting, testing, building on PRs and not every push, I guess we have to move the building image part into a separate workflow (that gets only triggered on push to `main`)

---

We decided to only build a `latest` image on the `main`-Branch for now.  In PRs no images are created.